File: Data Structures & Algorithms/string-encode-and-decode/submission-1.py
class Solution:
    def Double1(self, s: str):
        ans = ""
        for i in s:
            ans += str(i)*2
        return ans

    def encode(self, strs: List[str]) -> str:
        ans = ""
        for i in strs:
            ans += str(self.Double1(i))
            ans += "@/"
        return ans

    def decode(self, s: str) -> List[str]:
        l = []
        word = ""

        i = 0
        while(i < len(s)-1):
            if(s[i] == s[i+1]):
                word += s[i]
            else:
                l.append(word)
                word = ""
            i+= 2
        if (word != ""):
            l.append(word)
        return l

File: Data Structures & Algorithms/reverse-bits/submission-1.cpp
int f(uint32_t n){
    int ans = 0;
    int l = 0;
    while(n>0){
        ans*= 2;
        if(n%2==1)ans++;
        n=n/2;
        l++;
    }
    cout << l << endl;
    for(int i = 0; i < 32-l; i++)ans*=2;
    return ans;
}

class Solution {
public:
    uint32_t reverseBits(uint32_t n) {
        return f(n);
    }
};
